File: week_3_service_implementation/hello-go/hello.go
package main

import (
	"fmt"
	"net/http"
)

func main() {

	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Add("Content-Type", "text/html")
		w.WriteHeader(http.StatusOK)
		fmt.Fprintf(w, "Hello from Go!")
	})

	fmt.Printf("Starting Go server at port 8080\n")
	http.ListenAndServe(":8080", nil)
}

File: week_3_service_implementation/hello/hello-js/hello.js
// load modules
const http = require("http");

// code
console.log("Starting NodeJS server at port 9090");

function handleRoot(req, res) {
  res.writeHead(200, { "Content-Type": "text/html" });
  res.write("Hello from NodeJS!");
  res.end();
}

http
  // handle path "/"
  .createServer(handleRoot)
  .listen(9090);
